### PR TITLE
feat: Updated Doodle MultiSelect style variants - BED-9220

### PR DESCRIPTION
**Pull request by nicole-m-martin**
## Description

`Doodle MultiSelect` component currently appears filled whenever values are selected, even when they match the configured default. The new variant prop allows consumers to use an outlined default state without changing the component’s existing behavior.

- Added optional `outlined` and `filled` trigger variants to the `Doodle MultiSelect.`
- Added Storybook documentation and test coverage for the new variants.

## Motivation and Context

Resolves: [BED-9220](https://specterops.atlassian.net/browse/BED-9220)

## How Has This Been Tested?
- Local manual testing.
- Testing in Storybook.
- New test added to multiSelect.

## Types of changes

- Chore (a change that does not modify the application functionality)
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to change)
- Database Migrations

## Checklist:

- [x] I have met the contributing prerequisites
  - Assigned myself to this PR
  - Added the appropriate labels
  - Associated an issue: https://github.com/SpecterOps/BloodHound/issues/672
  - Read the Contributing guide: https://github.com/SpecterOps/BloodHound/wiki/Contributing
- [ ] I have ensured that related documentation is up-to-date
  - Open API docs
  - Code comments (GoDocs / JSDocs)
- [x] I have followed proper test practices
  - Added/updated tests to cover my changes
  - All new and existing tests passed

[BED-9220]: https://specterops.atlassian.net/browse/BED-9220?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## Summary by CodeRabbit

* **New Features**
  * Added outlined and filled visual variants for the MultiSelect component.
  * MultiSelect can now automatically select an appropriate style based on its state, with optional manual control.
  * Exposed whether a multi-value filter matches its default selection.

* **Documentation**
  * Added examples demonstrating both MultiSelect variants and their intended usage.

* **Tests**
  * Added coverage confirming the outlined MultiSelect styling is applied correctly.

### CODE REVIEW

File: packages/javascript/doodle-ui/src/components/MultiSelect/MultiSelect.test.tsx [hunk 1]
```diff
@@ -305,4 +305,24 @@ describe('MultiSelect', () => {
...
+        );
+
+        const trigger = screen.getByRole('button', {
+            name: /menu item a/i,
+        });
+
+        expect(trigger).toHaveClass('bg-select-trigger-outlined-fill');
+        expect(trigger).toHaveClass('ring-1');
+        expect(trigger).not.toHaveClass('bg-primary');
+    });
```
Comment (bfaulk96) [inline packages/javascript/doodle-ui/src/components/MultiSelect/MultiSelect.test.tsx:327]: Would it be useful to also have a test for showing the filled state still for `variant='outlined'` when a non-empty state outside of the default is selected?

File: packages/javascript/doodle-ui/src/components/MultiSelect/MultiSelect.tsx [hunk 2]
```diff
@@ -38,13 +38,31 @@ const CaretDown = ({ className, size = 12 }: { className?: string; size?: number
     </svg>
 );
 
+type MultiSelectVariant = 'outlined' | 'filled';
```
Comment (bfaulk96) [inline packages/javascript/doodle-ui/src/components/MultiSelect/MultiSelect.tsx:41]: ~~I kind of feel like `outlined` vs `filled` is misleading, might make more sense to have an attribute like `defaultState: 'all' | 'none'`? Or `variant: 'all' | 'none'`? Something that just captures intent better? What are your thoughts on that?~~

**Review by bfaulk96 — APPROVED** [review]
Two nit-picks/questions, but seems like a solid change!

File: packages/javascript/doodle-ui/src/components/MultiSelect/MultiSelect.tsx (hunk 2 above)
Comment (bfaulk96) [reply packages/javascript/doodle-ui/src/components/MultiSelect/MultiSelect.tsx:41]: Ah, while reviewing the BHE PR, I now see the goal for how these are being used. Disregard!

File: packages/javascript/doodle-ui/src/components/MultiSelect/MultiSelect.tsx [hunk 3]
```diff
@@ -190,6 +215,9 @@ const MultiSelect = ({
         }
     };
 
+    const triggerVariant: MultiSelectVariant =
+        variant ?? (value.length === 0 && !disabled && !error ? 'outlined' : 'filled');
```
Comment (dcairnsspecterops) [inline packages/javascript/doodle-ui/src/components/MultiSelect/MultiSelect.tsx]: This is a cool line of code ⛳, but I might want to see it split into a few lines for readability. Either way!

**Review by dcairnsspecterops — APPROVED** [review]

File: packages/javascript/doodle-ui/src/components/MultiSelect/MultiSelect.test.tsx (hunk 1 above)
Comment (nicole-m-martin) [reply packages/javascript/doodle-ui/src/components/MultiSelect/MultiSelect.test.tsx:327]: Yes! Added test in the FindingsTableFilters file in BHE. Good call

File: packages/javascript/doodle-ui/src/components/MultiSelect/MultiSelect.tsx (hunk 3 above)
Comment (nicole-m-martin) [reply packages/javascript/doodle-ui/src/components/MultiSelect/MultiSelect.tsx]: updated. :)
